apps/fz_http/lib/fz_http/configurations/mailer.ex

```diff
@@ -0,0 +1,29 @@
+defmodule FzHttp.Configurations.Mailer do
+  @moduledoc """
+  A non-persisted schema to validate email configs on boot.
+  XXX: Consider persisting this to make outbound email configurable via API.
+  """
+
+  defstruct [:from, :provider, :configs]
+  @types %{from: :string, provider: :string, configs: :map}
+
+  def changeset(attrs) do
+    {%__MODULE__{}, @types}
```

Comment:
My unsolicited feedback,

```suggestion
  use Ecto.Schema
  
  embedded_schema do
    field :from, :string
    field :provider, :string
    field :configs, :string # or embeds_one :configs, Config
  end

  def changeset(attrs) do
    %__MODULE__{}
```

I think it is worth leveraging Ecto `embedded_schema` for these situations. We have been leveraging `Ecto.Schema` for all Value Objects, which has worked well. In the end, Ecto is not about databases per se.

Reply:
Hey @yordis! Appreciate the feedback. `embedded_schema` is what I was after, thanks for the catch.

Reply:
Beautiful! Joyful to see that it was helpful 💜